src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -509,14 +522,30 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
       callOutputs foreach updateSymbolCache(callKey)
       logger.debug(s"In state WorkflowRunning: Got PersistenceCompleted message for Done call ${callKey.tag}")
       val updatedData = startRunnableCalls(data)
-      if (isWorkflowDone) scheduleTransition(WorkflowSucceeded)
       val finalData = updatedData.removePersisting(callKey, ExecutionStatus.Done)
+      self ! CheckForWorkflowComplete
       stay using finalData
     case Event(PersistenceSucceeded(callKey, ExecutionStatus.Preempted, callOutputs), data) =>
       executionStore += callKey -> ExecutionStatus.Preempted
       val updatedData = startRunnableCalls(data)
       val finalData = updatedData.removePersisting(callKey, ExecutionStatus.Preempted)
+      self ! CheckForWorkflowComplete
       stay using finalData
+    case Event(PersistenceSucceeded(callKey, ExecutionStatus.Failed, callOutputs), data) =>
+      executionStore += callKey -> ExecutionStatus.Failed
+      val finalData = data.removePersisting(callKey, ExecutionStatus.Failed)
+      self ! CheckForWorkflowComplete
+      stay using finalData
+    case Event(CheckForWorkflowComplete, data) =>
+      logger.info("Checking for workflow complete! executuib")
+      if (!(executionStore exists ( (p: (ExecutionStoreKey, ExecutionStatus)) => p._2 == ExecutionStatus.Running)) && data.pendingExecutions.isEmpty) {
```

Comment:
If this persists beyond Miguel's below comment - did you swap a pair of {}'s for ()'s? Also the opening ( has a space and the closing ) does not (here and below)